frontend/src/metabase/querying/utils/drills/query-drill.ts

```diff
@@ -18,6 +18,12 @@ export function queryDrill(
     clicked?.dimensions,
   );
 
+  const isDashboard = clicked?.extraData?.dashboard != null;
+
+  if (isDashboard) {
+    question = question.lockDisplay();
+  }
```

Comment:
Consider this:
1. Create a question [like this](https://github.com/metabase/metabase/assets/6830683/5ff06e5c-3f7c-4c91-a7b0-193fa7a6a832), visualize it as pie chart, save it and add it to a dashboard
2. Now in the dashboard, click on data in the pie chart, select "Break out by..." -> "Location" -> "State"

In `master`, this will result in a nice [US map](https://github.com/metabase/metabase/assets/6830683/c3829281-0ac2-41be-a67c-bc03d10c84db).
With this change, we are presented with a [pie chart](https://github.com/metabase/metabase/assets/6830683/1444420a-526c-4f8c-ab8c-f50b6ae1ac92) - definitely 

From this we can conclude that it's not desired to simply always lock the display.

----

The condition when to lock the display will have to be more sophisticated.

Reply:
> When doing a zoom from a Question, the display was already locked.

This may be incorrect, as I would expect the above scenario to yield a US map both in dashboard and a question view.

Reply:
Ah this is a great point thanks. 
I will reconsider my approach and have a closer look at my assumptions about locking display in questions too.

Reply:
@kamilmielnik I tried your example on Stats but was unable to get the map. Any time I did the zoom in it still brought me to another pie chart. Could you perhaps show a quick screen recording? Or some more screenshots showing exactly how you achieved that?

Reply:
@adam-james-v this is from yesterday's `master`:

https://github.com/metabase/metabase/assets/6830683/2c227064-95f6-48bd-b6c1-8f896ac98e02

Reply:
Nice! I like the simplicity of the solution.

I could not come up with any counter-examples to this scenario but it does not mean they're not there.
Maybe it'd be a good idea to get product team's approval before we merge this change.

I have withdrawn my rejection.
It looks like the test this PR adds is failing in CI.